Trendyol.App.WebApi/Filters/BasicAuthenticationFilter.cs

```diff
@@ -0,0 +1,151 @@
+
+            Tuple<string, string> userNameAndPasword = ExtractUserNameAndPassword(authorization.Parameter);
+            if (userNameAndPasword == null)
+            {
+                context.ErrorResult = new AuthenticationFailureResult("Invalid credentials", request);
+                return;
+            }
+
+            string userName = userNameAndPasword.Item1;
+            string password = userNameAndPasword.Item2;
+
+            IPrincipal principal = null;
+            if (_checker.Check(userName, password, cancellationToken))
+            {
+                GenericIdentity identity = new GenericIdentity(userName);
+                principal = new GenericPrincipal(identity, new List<string>().ToArray());
+            }
+
+            if (principal == null)
+            {
+                context.ErrorResult = new AuthenticationFailureResult("Invalid username or password", request);
+                return;
+            }
+
+            context.Principal = principal;
+        }
+
+
+        public Task ChallengeAsync(HttpAuthenticationChallengeContext context, CancellationToken cancellationToken)
+        {
+            AuthenticationHeaderValue challenge = new AuthenticationHeaderValue("Basic");
+            context.Result = new AddChallengeOnUnauthorizedResult(challenge, context.Result);
+            return Task.FromResult(0);
+        }
+
+
+        private Tuple<string, string> ExtractUserNameAndPassword(string authorizationParameter)
+        {
+            authorizationParameter = Encoding.Default.GetString(Convert.FromBase64String(authorizationParameter));
+
+            string[] tokens = authorizationParameter.Split(':');
+            return tokens.Length != 2
+                       ? null
+                       : new Tuple<string, string>(tokens[0], tokens[1]);
+        }
+
+
+        private class AddChallengeOnUnauthorizedResult : IHttpActionResult
+        {
+            public AddChallengeOnUnauthorizedResult(AuthenticationHeaderValue challenge, IHttpActionResult innerResult)
+            {
+                Challenge = challenge;
+                InnerResult = innerResult;
+            }
+
+            public AuthenticationHeaderValue Challenge { get; private set; }
+
+            public IHttpActionResult InnerResult { get; private set; }
+
+            public async Task<HttpResponseMessage> ExecuteAsync(CancellationToken cancellationToken)
+            {
+                HttpResponseMessage response = await InnerResult.ExecuteAsync(cancellationToken);
+
+                if (response.StatusCode == HttpStatusCode.Unauthorized)
+                {
+                    if (response.Headers.WwwAuthenticate.All(h => h.Scheme != Challenge.Scheme))
+                    {
+                        response.Headers.WwwAuthenticate.Add(Challenge);
+                    }
+                }
+
+                return response;
+            }
+        }
+
+        private class AuthenticationFailureResult : IHttpActionResult
```

Comment:
This result also need to be shaped according to our standards those followed on API response models.
You need to return an object derived from BaseResponse to begin with.